openpgp/armor/armor.go

```diff
@@ -118,8 +116,11 @@ func (l *lineReader) Read(p []byte) (n int, err error) {
 		return 0, io.EOF
 	}
 
-	if len(line) > 96 {
-		return 0, ArmorCorrupt
+	if bytes.HasPrefix(line, armorEnd) {
```

Comment:
This + removing `isPrefix` check on line 87 has a side effect of being able to bail out in the middle of a line, if it `armorEnd` happens to align with bufio buffer.